version: '3.10'

volumes:
  pgdata:

networks:
  dev_network:

services:

  pg_db:
    image: postgres:latest
    container_name: retail_orders_pg_db
    environment:
      POSTGRES_DB: ${DB_NAME}
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    ports:
      - '${DB_HOST_PORT}:${DB_PORT}'
    volumes:
      - pgdata:/var/lib/postgresql/data
    networks:
      - dev_network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}" ]
      interval: 30s
      timeout: 30s
      retries: 3
      start_period: 20s
    restart: unless-stopped
    deploy:
      resources:
        limits:
          cpus: "0.5"
          memory: 1G

  pgadmin:
    image: dpage/pgadmin4:latest
    ports:
      - "5050:80"
    container_name: pgadmin
    depends_on:
      - pg_db
    environment:
      PGADMIN_DEFAULT_EMAIL: "aaa@yandex.ru"
      PGADMIN_DEFAULT_PASSWORD: "111"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - ./services/pgadmin:/var/lib/pgadmin
    networks:
      - dev_network
    deploy:
      resources:
        limits:
          cpus: "0.5"
          memory: 1G

  redis:
    image: redis:7.0.0-alpine3.15
    ports:
      - "${REDIS_HOST_PORT}:${REDIS_PORT}"
    networks:
      dev_network:
        aliases:
          - redis_db
    deploy:
      resources:
        limits:
          cpus: "0.5"
          memory: 512M

#  celery:
